api/views/student.py

```diff
@@ -105,6 +109,38 @@ def missed_class(self, request: Request, personal_info_id: int) -> Response:  #
         )
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        parameters=[DashboardAvailableStudentsSerializer],
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                response=ListSerializer(child=MinifiedStudentSerializer()),
+                description="Students returned",
+            ),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Something is wrong with the query params",
+            ),
+            status.HTTP_409_CONFLICT: OpenApiResponse(
+                response=BaseAPIExceptionSerializer,
+                description="Invalid time slots",
+            ),
```

Comment:
I think that `400 BAD REQUEST` should be used for both cases, we should give exact reason in the response body. If the dashboard gives wrong IDs or only one slot instead of two, it's definitely a `BAD REQUEST` IMO. `409 CONFLICT` could be returned when no students are found, but IIRC we voted and decided that `200 OK` be returned with an empty list in such cases.

Reply:
Here probably the message is confusing, but I can change the message to "some of the time slots not found" or change the status as well. I used 409 everywhere where some of the body items were not found but here we have query param so it might be reasonable to use `not found`

Reply:
Okay, according to [this vote](https://t.me/c/1808422267/1258), in a similar case we decided to return 200 OK and empty list, so let's do the same here.

Reply:
I see that throughout our code `409` is used for `POST` and `DELETE` requests, and I think this is consistent with recommended practice. So it was the right call to get rid, but I'm not sure about `404`. I guess we'll continue this discussion under [the newer comment](https://github.com/samanthasgroup/django-webapps/pull/280#discussion_r1349576910).